source-container-build/app/source_build.py

```diff
@@ -417,6 +422,49 @@ def generate_source_images(image: str) -> list[str]:
     return [deprecated_image, source_image]
 
 
+def resolve_source_image_by_version_release(binary_image: str) -> str | None:
+    """Resolve source image by inspecting version and release of binary image
+
+    :param binary_image: str, resolve source image for this binary image. A valid image reference
+        should be passed, which then is inspected from the specified registry.
+    :return: the resolved source image. If no source image is resolved, None is returned.
+    """
+    log = logging.getLogger(f"{logger.name}.resolve_source_image")
+    # Expected form: version-release
+    image_config = fetch_image_config(binary_image)
+    config_data = json.loads(image_config)
+    version = config_data["config"]["Labels"].get("version")
+    release = config_data["config"]["Labels"].get("release")
+    if not (version and release):
+        log.warning("Image %s is not labelled with version and release.", binary_image)
+        return
+    # Remove possible tag or digest from binary image
+    image_name = binary_image.split("@")[0]
+    image_name = image_name.rsplit(":", 1)[0]
+    source_image = f"{image_name}:{version}-{release}-source"
+    if registry_has_image(source_image):
+        return source_image
+
+
+def drop_tag(image: str) -> str:
+    """Drop tag from an image pullspec if it includes both tag and digest
+
+    :param image: str, image pullspec.
+    :return: tag is removed from the given image pullspec and the result is returned.
+    :rtype: str
+    """
+    parts = image.split("@")
+    digest = ""
+    if len(parts) == 2:
+        digest = parts[1]
+    parts = parts[0].rsplit(":", 1)
+    has_tag = len(parts) == 2 or (len(parts) > 2 and "/" in parts[0])
+    if digest and has_tag:
+        # Drop the tag
+        return parts[0] + "@" + digest
```

Comment:
Does `parts[0] + "@" + digest` correctly handle the case where `len(parts) > 2`?

Reply:
Dropped `drop_tag` and replaced it with a rough image name parser so that the parsed result can be reused, at where your [comment](https://github.com/redhat-appstudio/build-tasks-dockerfiles/pull/33#discussion_r1482723897) mentioned.